src/record_mixing.c

```diff
@@ -25,33 +25,46 @@
 #include "task.h"
 #include "tv.h"
 
+extern u8 ewram[];
+#define unk_2018000 (*(struct PlayerRecords *)(ewram + 0x18000))
+#define unk_2008000 (*(struct PlayerRecords *)(ewram + 0x08000))
+
+extern struct RecordMixing_UnknownStruct gUnknown_02038738;
+
+extern void *const recordMixingSecretBases;
+extern void *const recordMixingTvShows;
+extern void *const gUnknown_083D0274;
+extern void *const gUnknown_083D0278;
+extern void *const recordMixingEasyChatPairs;
+extern struct RecordMixing_UnknownStruct *const gUnknown_083D0280;
+extern void *const gUnknown_083D0284;
 
-extern void *recordMixingSecretBases;
-extern void *recordMixingTvShows;
-extern void *gUnknown_083D0274;
-extern void *gUnknown_083D0278;
-extern void *recordMixingEasyChatPairs;
-extern void *gUnknown_083D0284;
-extern u8 gUnknown_083D0288[2];
-extern u8 gUnknown_083D028A[2][3];
-extern u8 gUnknown_083D0290[9][4];
-
-extern struct RecordMixing_UnknownStruct gUnknown_02038738;  //Don't know what type this points to
-extern struct RecordMixing_UnknownStruct *gUnknown_083D0280;
 extern u16 gSpecialVar_0x8005;
 extern u32 gUnknown_03005D2C;
 extern u8 gUnknown_03000718;
 extern u8 gUnknown_0300071C[];
 extern bool8 gReceivedRemoteLinkPlayers;
 
+// I can't define these here or else RecordMixing_PrepareExchangePacket gets optimized.
+/*
+void *const recordMixingSecretBases = &gSaveBlock1.secretBases;
+void *const recordMixingTvShows = &gSaveBlock1.tvShows;
+void *const gUnknown_083D0274 = &gSaveBlock1.unknown_2ABC;
+void *const gUnknown_083D0278 = &gSaveBlock1.oldMan;
+void *const recordMixingEasyChatPairs = &gSaveBlock1.easyChatPairs;
+struct RecordMixing_UnknownStruct *const gUnknown_083D0280 = &gUnknown_02038738;
+void *const gUnknown_083D0284 = &gSaveBlock2.filler_A8;
+*/
```

Comment:
No, that doesn't match. I also tried a struct, and it does the same thing. What does match is if I uncomment those variables and remove the `const`. However, that doesn't build because it gets put into `.data` instead of `.rodata`. I have a feeling that the original linker script put `.data` into ROM along with `.rodata`, and Game Freak didn't use `const` on every single piece of ROM data.